Make the Android pages more consistent (#1086)
privacyguides/privacyguides.org: docs/android/grapheneos-vs-calyxos.en.md

## Sandboxed Google Play vs Privileged microG

When Google Play services are used on GrapheneOS, they run as a user app and are contained within a user or work profile.

Sandboxed Google Play is confined using the highly restrictive, default [`untrusted_app`](https://source.android.com/security/selinux/concepts) domain provided by [SELinux](https://en.wikipedia.org/wiki/Security-Enhanced_Linux). Permissions for apps to use Google Play Services can be revoked at any time by the user.

microG is an open-source re-implementation of Google Play Services. This means it needs to be updated every time Android has a major version update (or the Android API changes). It also needs to run in the highly privileged [`system_app`](https://source.android.com/security/selinux/concepts) SELinux domain like regular Google Play Services, and it requires an operating system that allows [signature spoofing](https://github.com/microg/GmsCore/wiki/Signature-Spoofing), which allows system apps to insecurely masquerade as other apps. This is less secure than Sandboxed Google Play's approach, which does not need access to sensitive system APIs.

> When using Sandboxed Play Services, you have the option to reroute location requests to the Play Services API back to the OS location API which uses satellite based location services. With microG, you have the option to either not use a network location backend at all, or to *shift trust* to another location backend like Mozilla or Dejavu. Using an alternative location backend still submits location data to these third-party providers, just not Google's own location services.

Reviewer: ```suggestion
When using Sandboxed Play Services, you have the option to reroute location requests to the Play Services API back to the OS location API which uses satellite based location services. With microG, you have the option to either not use a network location backend at all, *shift trust* to another location backend like Mozilla, or use [DejaVu](https://github.com/n76/DejaVu), a location backend that locally collects and saves RF-based location data to an offline database which can be used when GPS is not available.
```

GrapheneOS extends the user profile feature allowing a user to press an "End Session" button. This button clears the encryption key from memory. There are plans to add a [cross profile notifications system](https://github.com/GrapheneOS/os-issue-tracker/issues/88) in the future. GrapheneOS plans to introduce nested profile support with better isolation in the future.

## Sandboxed Google Play vs Privileged microG

When Google Play services are used on GrapheneOS, they run as a user app and are contained within a user or work profile.

Sandboxed Google Play is confined using the highly restrictive, default [`untrusted_app`](https://source.android.com/security/selinux/concepts) domain provided by [SELinux](https://en.wikipedia.org/wiki/Security-Enhanced_Linux). Permissions for apps to use Google Play Services can be revoked at any time by the user.

> microG is a reimplementation of Google Play Services. This means it needs to be updated every time Android has a major version update (or the Android API changes). It also needs to run in the highly privileged [`system_app`](https://source.android.com/security/selinux/concepts) SELinux domain like normal Google Play Services and requires access to [signature spoofing](https://madaidans-insecurities.github.io/android.html#microg-signature-spoofing) so this is less secure than the Sandboxed Google Play approach.

Reviewer: ```suggestion
microG is an open-source re-implementation of Google Play Services. This means it needs to be updated every time Android has a major version update (or the Android API changes). It also needs to run in the highly privileged [`system_app`](https://source.android.com/security/selinux/concepts) SELinux domain like regular Google Play Services, and it requires an operating system that allows [signature spoofing](https://github.com/microg/GmsCore/wiki/Signature-Spoofing), which allows system apps to insecurely masquerade as other apps. This is less secure than Sandboxed Google Play's approach, which does not need access to sensitive system APIs.
```

> When using Sandboxed Play Services, you have the option to reroute location requests to the Play Services API back to the OS location API which uses satellite based location services. With microG, you have the option to not use a network location backend or to *shift trust* to a location backend like Mozilla or Dejavu.
> 
> When threat modeling, a network location service like Mozilla is not that much better than Play Service location, since you will be still submitting Wifi and Bluetooth MAC addresses of the devices surrounding your device anyway. The Dejavu location backend, while being private, requires that you have a working GPS first for the local RF data collected to be useful. This implies that the phone can find out percisely where it is just fine without the need for such location service. If your threat model requires you to keep your location private from service providers, it is best that you rely soley on satellite based services.
> 
> The only threat model we can think of where you might want to use microG is one where you do not trust Google but instead trust Mozilla and other third parties with with the MAC addresses of the devices in your immediate location. We don't think this fits into a reasonable threat model worth pursuing.

Reviewer: ```suggestion
When using Sandboxed Play Services, you have the option to reroute location requests to the Play Services API back to the OS location API which uses satellite based location services. With microG, you have the option to either not use a network location backend at all, or to *shift trust* to another location backend like Mozilla or Dejavu. Using an alternative location backend still submits location data to these third-party providers, just not Google's own location services.
```

> When evaluating ease of use, we believe Sandboxed Google Play also works well with far more applications than microG, due to its support for services like [Google Play Games](https://play.google.com/googleplaygames) and [In-app Billing API](https://android-doc.github.io/google/play/billing/api.html). Authentication using [FIDO](security/multi-factor-authentication#fido-fast-identity-online) with online services on Android also relies on Play Services and the feature is not yet implemented in microG.

Reviewer: Not really an opinion, don't state it as such:
```suggestion
In terms of application compatibility, Sandboxed Google Play outperforms microG due to its support for many services which microG has not yet implemented, like [Google Play Games](https://play.google.com/googleplaygames) and [In-app Billing API](https://android-doc.github.io/google/play/billing/api.html). Authentication using [FIDO](security/multi-factor-authentication#fido-fast-identity-online) with online services on Android also relies on Play Services, and the feature is not yet implemented in microG.
```

When using Sandboxed Play Services, you have the option to reroute location requests to the Play Services API back to the OS location API which uses satellite based location services. With MicroG, you have the option to not use a network location backend or to *shift trust* to a location backend like Mozilla or Dejavu.

When threat modeling, a network location service like Mozilla is not that much better than Play Service location, since you will be still submitting Wifi and Bluetooth MAC addresses of the devices surrounding your device anyway. The Dejavu location backend, while being private, requires that you have a working GPS first for the local RF data collected to be useful. This implies that the phone can find out percisely where it is just fine without the need for such location service. If your threat model requires you to keep your location private from service providers, it is best that you rely soley on satellite based services.

> The only threat model we can think of where you might want to use MicroG is one where you do not trust Google but instead trust Mozilla and other third parties with with the MAC addresses of the devices in your immediate location. We don't think this fits into a reasonable threat model worth pursuing.

Reviewer: I would probably just remove these "when threat modeling" paragraphs entirely, they make no sense. This is not what threat modeling even is. _Clearly_ this threat model _does_ exist, and _we're not trying to tell people what their threat models are_, so this is a decision best left to the reader.

Reviewer 2: Agreed.

Reviewer: https://github.com/privacyguides/privacyguides.org/pull/1086#discussion_r856941910

Reviewer 2: I do think that fixes things.

When evaluating ease of use, we believe Sandboxed Google Play also works well with far more applications than MicroG, due to its support for services like [Google Play Games](https://play.google.com/googleplaygames) and [In-app Billing API](https://android-doc.github.io/google/play/billing/api.html). Authentication using [FIDO](security/multi-factor-authentication#fido-fast-identity-online) with online services on Android also relies on Play Services and the feature is not yet implemented in MicroG.

> 
> In conclusion, we do not believe MicroG provides any privacy or usability advantages over Sandboxed Google Play.

Reviewer: Adding a conclusion statement just makes it sound like you don't think the reader could grasp whatever you just said.

```suggestion
```

> Omit these phrases. “In conclusion” or “To conclude” may be appropriate for an oral presentation, but in writing are considered redundant or overly mechanical.

https://www.lib.sfu.ca/about/branches-depts/slc/writing/organization/conclusions

Reviewer 2: Agreed, and it also shouldn't be required if the sentence is succinct.
